### PR TITLE
WIP: API: Add support for PEP 519 / file system path protocol

### LINKED ISSUE #13823: API: Add support for PEP 519
File System Path Protocol has been accepted and implemented into Python 3.6, a backwards compatibility layer to support the introduced PEP 519 for python version < 3.6.

### PR DESCRIPTION
**Pull request by AbsoluteMSTR**
- [ ] closes #13823
- [ ] tests added / passed
- [ ] passes `git diff upstream/master | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by jreback** [comment]
needs tests

**Comment by jorisvandenbossche** [comment]
@AnkurDedania Do you have time to update this and add tests?

**Comment by jorisvandenbossche** [comment]
@AnkurDedania Do you have time to update this and add tests?

**Comment by jreback** [comment]
can you rebase / update

**Comment by AnkurDedania** [comment]
TIL how to rebase, if done correctly

**Comment by jorisvandenbossche** [comment]
It seems the rebase worked! 
Can you add some tests?

File: pandas/io/common.py [hunk 1]
```diff
@@ -183,6 +190,10 @@ def _stringify_path(filepath_or_buffer):
     -------
     str_filepath_or_buffer : a the string version of the input path
     """
+    if _OS_FSPATH_INSTALLED and isinstance(filepath_or_buffer, PathLike):
+        return fspath(filepath_or_buffer)
+    elif hasattr(filepath_or_buffer, "__fspath__"):
+        return filepath_or_buffer.__fspath__()
     if _PATHLIB_INSTALLED and isinstance(filepath_or_buffer, pathlib.Path):
```
Comment (jreback) [inline pandas/io/common.py]: might as well make the others ``elif``

**Review by jreback — COMMENTED** [review]
please add a whatsnew
and some tests (look for where we tests for PATHLIB and PYPATH), do the same

**Comment by jreback** [comment]
also need to add a ``skip_if_no_fspath`` in ``pandas.util.testing`` (again alongside pathlib, py.path ones)

**Comment by jreback** [comment]
@AnkurDedania ca you rebase / update

**Comment by jreback** [comment]
@AnkurDedania something went wrong.

try

```
git rebase -i origin/master
```

then
```
git push yourremote yourbrach -f
```

**Comment by jreback** [comment]
closing, you can cherry-pick your changes on top of master. ps open a new PR.

**Pull request closed without merging**